### PR TITLE
Add support for Nightwatch v3

### LINKED ISSUE #5: Example for adding custom commands is needed
I need an example for adding custom commands in a TypeScript way. 
It's missing.
Thanks.

### PR DESCRIPTION
**Pull request by garg3133**
Also fixes #5.

### CODE REVIEW

File: test/iFrame.ts [hunk 1]
```diff
@@ -4,8 +4,7 @@ const home: NightwatchTests = {
   'iFrame test': async () => {
     const iFrame = browser.page.IFrame();
     iFrame.navigate();
-    const frame = await browser.findElement(iFrame.elements.iframe);
-    browser.frame(frame);
+    browser.frame(iFrame.elements.iframe.selector);
```
Comment (vaibhavsingh97) [inline test/iFrame.ts:7]:
```suggestion
    browser.frame('@iframe');
```

**Review by vaibhavsingh97 — CHANGES_REQUESTED** [review]

File: nightwatch/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "allowJs": true,
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true
+  },
```
[the remark this answers is not in the document]
Comment (garg3133) [reply nightwatch/tsconfig.json:14]: This is to show that the tsconfig present in the `nightwatch` folder should be intact and not changes should be made to it unless absolutely necessary.

File: test/fileUpload.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ const home: NightwatchTests = {
     fileUpload
       .navigate()
       .uploadFile('@fileUploadInput', filePath)
-      .click('@submitButton')
+      .strictClick(fileUpload.elements.submitButton)
```
[the remark this answers is not in the document]
Comment (garg3133) [reply test/fileUpload.ts:12]: Just to show another way of passing the element. Probably should do it on an internal command and also add a comment.

File: test/iFrame.ts (hunk 1 above)
Comment (garg3133) [reply test/iFrame.ts:7]: ^ won't work. Dependent on https://github.com/nightwatchjs/nightwatch/pull/3819

File: nightwatch/pages/Dropdown.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+const dropdown = {
+  url(this: EnhancedPageObject) {
+    return `${this.api.launch_url}/dropdown`;
+  },
+  elements
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const _: PageObjectModel = dropdown;
```
[the remark this answers is not in the document]
Comment (garg3133) [reply nightwatch/pages/Dropdown.ts:15]: Just to show another way of writing page objects, as `satisfies` is only available from TypeScript 4.9. Probably should also add a FAQ section in README and list this question: "We're using TypeScript < 4.9 which does not have a `satisfies` operator, how should we define the page objects?"

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -15,7 +15,7 @@ jobs:
       name: nightwatch-browserstack-build
     strategy:
       matrix:
-        node-version: [14.x, 15.x, 16.x]
```
[the remark this answers is not in the document]
Comment (garg3133) [reply .github/workflows/build.yml:18]: I think we should keep it as long as the main Nightwatch repo is supporting 14.x

File: nightwatch/pages/Dropdown.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply nightwatch/pages/Dropdown.ts:15]: > should also add a FAQ section in README and list this question: "We're using TypeScript < 4.9 which does not have a satisfies operator, how should we define the page objects?"

Yes, a FAQ question sounds good to me.

[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply nightwatch/pages/Dropdown.ts:15]: Why are we defining the `PageObjectModel` separately when we can assign it directly while defining the page object?

File: nightwatch/tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply nightwatch/tsconfig.json:14]: yup, this makes sense

File: test/fileUpload.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply test/fileUpload.ts:12]: Yes, we can add a comment showing possible ways of doing it.

```typescript
   fileUpload
      .navigate()
      .uploadFile('@fileUploadInput', filePath)
      .click('@submitButton')
      // we can also specify selector with @<selector-name>
      //.strictClick('@submitbutton')
      .strictClick(fileUpload.elements.submitButton)
```

File: nightwatch/pages/Dropdown.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (garg3133) [reply nightwatch/pages/Dropdown.ts:15]: Explained in the comment.

File: test/fileUpload.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (garg3133) [reply test/fileUpload.ts:12]: Done

**Comment by garg3133** [comment]
@beatfactor @gravityvi Can you please review this as well?

**Comment by garg3133** [comment]
Tests are only failing because of some config issues with BrowserStack, there are working fine locally.

**Pull request merged**